File: Add_Digits.py
num = 0

def foo(num : int):
    plus = 0
    for i in range(10):
        add = [i for i in str(num)]
        for j in add:
            plus += int(j)
        if len(str(plus)) == 1:
            return plus
        num = plus
        plus = 0
        add.clear()
    return 0

print(foo(num))

File: Add_Digits.c
#include<stdio.h>

int Add_digit(int num);

int main()
{
    printf("%d", Add_digit(38));
}

int Add_digit(int num)
{
    int add = 0;
    while (1)
    {
        while (num > 0)
        {
            add += num%10;
            num /= 10;
        }
        if (add < 10)
            break;
        num = add;
        add = 0;

    }
    return add;
}